xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -657,15 +712,30 @@ void CGUIDialogPVRTimerSettings::AddCondition(
   setting->SetDependencies(deps);
 }
 
-int CGUIDialogPVRTimerSettings::GetDateAsInt(const CDateTime &datetime)
+int CGUIDialogPVRTimerSettings::GetDateAsIndex(const CDateTime &datetime)
 {
-  const CDateTime date(datetime.GetYear(), datetime.GetMonth(), datetime.GetDay(), 0, 0, 0);
   time_t t(0);
-  date.GetAsTime(t);
+  const CDateTime date(datetime.GetYear(), datetime.GetMonth(), datetime.GetDay(), 0, 0, 0);
+  if (date > m_maxDateCovered)
+    (m_maxDateCovered + CDateTimeSpan(1, 0, 0, 0)).GetAsTime(t);
+  else if (date < m_minDateCovered)
+    (m_minDateCovered - CDateTimeSpan(1, 0, 0, 0)).GetAsTime(t);
+  else
+    date.GetAsTime(t);
```

Comment:
I'm not sure whether this is clever or just makes things complicated witout a real value add. See my comment regarding the > and < entries feature as such.